testing/scenario_app/ios/Scenarios/ScenariosUITests/iPadGestureTests.m

```diff
@@ -62,23 +65,136 @@ - (void)testPointerButtons {
+  } else {
+    rightClickAddedSequenceNumber =
+        assertOneMessageAndGetSequenceNumber(messages, @"PointerChange.add,device=2,buttons=0");
+    rightClickDownSequenceNumber =
+        assertOneMessageAndGetSequenceNumber(messages, @"PointerChange.down,device=2,buttons=2");
+    rightClickUpSequenceNumber =
+        assertOneMessageAndGetSequenceNumber(messages, @"PointerChange.up,device=2,buttons=2");
+  }
+  XCTAssertGreaterThan(rightClickDownSequenceNumber, rightClickAddedSequenceNumber,
+                       @"Right-click pointer was pressed before it was added");
+  XCTAssertGreaterThan(rightClickUpSequenceNumber, rightClickDownSequenceNumber,
+                       @"Right-click pointer was released before it was pressed");
+  XCTAssertGreaterThan([[hoverSequenceNumbers firstObject] intValue], 3,
+                       @"Hover occured before hover pointer was added");
+  XCTAssertGreaterThan(hoverRemovedSequenceNumber, [[hoverSequenceNumbers lastObject] intValue],
+                       @"Hover occured after hover pointer was removed");
+}
+
+- (void)testPointerHover {
+  BOOL supportsPointerInteraction = NO;
+  SEL supportsPointerInteractionSelector = @selector(supportsPointerInteraction);
+  if ([XCUIDevice.sharedDevice respondsToSelector:supportsPointerInteractionSelector]) {
+    supportsPointerInteraction =
+        performBoolSelector(XCUIDevice.sharedDevice, supportsPointerInteractionSelector);
+  }
+  XCTSkipUnless(supportsPointerInteraction, "Device does not support pointer interaction.");
+  XCUIApplication* app = [[XCUIApplication alloc] init];
+  app.launchArguments = @[ @"--pointer-events" ];
+  [app launch];
+
+  NSPredicate* predicateToFindFlutterView =
+      [NSPredicate predicateWithFormat:@"identifier BEGINSWITH 'flutter_view'"];
+  XCUIElement* flutterView = [[app descendantsMatchingType:XCUIElementTypeAny]
+      elementMatchingPredicate:predicateToFindFlutterView];
+  if (![flutterView waitForExistenceWithTimeout:kSecondsToWaitForFlutterView]) {
+    NSLog(@"%@", app.debugDescription);
+    XCTFail(@"Failed due to not able to find any flutterView with %@ seconds",
+            @(kSecondsToWaitForFlutterView));
+  }
+
+  XCTAssertNotNil(flutterView);
+
+  SEL hover = @selector(hover);
+  XCTAssertTrue([flutterView respondsToSelector:hover],
+                @"If supportsPointerInteraction is true, this should be true too.");
+  [flutterView performSelector:hover];
+  XCTAssertTrue(
+      [app.textFields[@"0,PointerChange.add,device=0,buttons=0"] waitForExistenceWithTimeout:1],
+      @"PointerChange.add event did not occur for a hover");
+  XCTAssertTrue(
+      [app.textFields[@"1,PointerChange.hover,device=0,buttons=0"] waitForExistenceWithTimeout:1],
+      @"PointerChange.hover event did not occur for a hover");
+  // The number of hover events fired is not always the same
+  NSInteger lastHoverSequenceNumber = -1;
```

Comment:
Initializing to `NSNotFound` and checking `(lastHoverSequenceNumber == NSNotFound) || (messageSequenceNumber > lastHoverSequenceNumber)` is probably more idiomatic than -1, but this is fine.